### PR TITLE
Improve handling of spacy dependency & fix audio-transcriptions demo

### PR DESCRIPTION
**Pull request by aaron-siegel**
Improves handling of spacy dependency by "inlining" the model download in `env.py`, removing the need to specify the model as a separate dependency. Fixes the audio-transcriptions demo by adding spacy as a separate dependency.

TESTING DONE: ran the audio-transcriptions demo (1) on colab, and (2) locally in a clean conda environment, verifying that the dependencies are installed correctly.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/289)

File: pixeltable/env.py [hunk 1]
```diff
@@ -511,9 +514,29 @@ def __register_packages(self) -> None:
         self.__register_package('whisperx')
         self.__register_package('yolox', library_name='git+https://github.com/Megvii-BaseDetection/YOLOX@ac58e0a')
 
-        if self.is_installed_package('spacy'):
-            import spacy
-            self._spacy_nlp = spacy.load('en_core_web_sm')
+    def __init_spacy(self) -> None:
```
Comment (mkornacker) [inline pixeltable/env.py]: comment: what are we doing here and why is this necessary

**Review by mkornacker — APPROVED** [review]

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/289)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @mkornacker)

___
*[`pixeltable/env.py` line 517 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/289#gh-1777604012:-O7juN1X91quOM__i_T8:bqmjwcz) ([raw file](https://github.com/pixeltable/pixeltable/blob/2c01a2d0f221f3a43b94c5d3f980dbcfbe9156da/pixeltable/env.py#L517)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

comment: what are we doing here and why is this necessary
</blockquote></details>

Added.
